### DISCUSSION TITLE
Adapt to SD-v3's VAE

### DISCUSSION THREAD

**Discussion by k-zha14** [PixArt-alpha/PixArt-alpha#155] [Q&A]
Hi, great work! I succeed in reproducing the VAE adaption from SD2' to SDXL's, as discussed in Pixart-Sigma. However, the adaption to SDXL's VAE is not successful. After 10k steps finetuning in SAM, sampled images are meaningless and chaotic(attached below), although the training loss looks pretty good.

![image (3)](https://github.com/PixArt-alpha/PixArt-alpha/assets/32434681/783b55ab-6e26-4b06-a5e1-c29265d7d786)
![image (2)](https://github.com/PixArt-alpha/PixArt-alpha/assets/32434681/396c121b-316f-4e5d-a552-76271551c2e1)
![image (1)](https://github.com/PixArt-alpha/PixArt-alpha/assets/32434681/7261405e-f6e6-40f9-9dd2-6d094bee110e)
![image](https://github.com/PixArt-alpha/PixArt-alpha/assets/32434681/9f8f2fd5-52cc-4ffb-8d79-5231cbda249a)

(the first two images generated from the the adaption exp. - SD2's to SDXL's VAE, while the latter from SD2' to SDv3's VAE)

The key change of SDv3's VAE is, the latent channel expands from 4 to 16. Thus, the compressed latents will reserve more details and avoid unplesant artifacts(eg. little faces, texts). To accommodate with this change, I initialize the net with official 'Pixart-alpha-256x256.pt' weights, except the 'x_embed' layer and 'final_layer'(channels 4->16, 8 -> 32).

Could anyone give me some hints? I'm really confused. Thanks, guys!
